test/unit/BPool.t.sol

```diff
@@ -326,41 +326,145 @@ contract BPool_Unit_GetController is BasePoolTest {
 }
 
 contract BPool_Unit_SetSwapFee is BasePoolTest {
-  function test_Revert_Finalized() private view {}
+  function test_Revert_Finalized() public {
+    _setFinalize(true);
 
-  function test_Revert_NotController() private view {}
+    vm.expectRevert('ERR_IS_FINALIZED');
+    bPool.setSwapFee(0);
+  }
+
+  function test_Revert_NotController(address _controller, address _caller) public {
+    vm.assume(_controller != _caller);
+    bPool.set__controller(_controller);
 
-  function test_Revert_MinFee() private view {}
+    vm.expectRevert('ERR_NOT_CONTROLLER');
+    vm.prank(_caller);
+    bPool.setSwapFee(0);
+  }
 
-  function test_Revert_MaxFee() private view {}
+  function test_Revert_MinFee(uint256 _fee) public {
+    vm.assume(_fee < MIN_FEE);
 
-  function test_Revert_Reentrancy() private view {}
+    vm.expectRevert('ERR_MIN_FEE');
+    bPool.setSwapFee(_fee);
+  }
 
-  function test_Set_SwapFee() private view {}
+  function test_Revert_MaxFee(uint256 _fee) public {
+    vm.assume(_fee > MAX_FEE);
 
-  function test_Emit_LogCall() private view {}
+    vm.expectRevert('ERR_MAX_FEE');
+    bPool.setSwapFee(_fee);
+  }
+
+  function test_Revert_Reentrancy() public {
+    // Assert that the contract is accesible
+    assertEq(bPool.call__mutex(), false);
+
+    // Simulate ongoing call to the contract
+    bPool.set__mutex(true);
+
+    vm.expectRevert('ERR_REENTRY');
+    bPool.setSwapFee(0);
+  }
+
+  function test_Set_SwapFee(uint256 _fee) public {
+    vm.assume(_fee >= MIN_FEE);
+    vm.assume(_fee <= MAX_FEE);
+
+    bPool.setSwapFee(_fee);
+
+    assertEq(bPool.call__swapFee(), _fee);
+  }
+
+  function test_Emit_LogCall(uint256 _fee) public {
+    vm.assume(_fee >= MIN_FEE);
+    vm.assume(_fee <= MAX_FEE);
```

Comment:
Let's create a `happyPath` modifier for `ERR_MIN_FEE` and `ERR_MAX_FEE` not to happen.